### ISSUE TITLE
Switch from ISO-8601 format to UNIX timestamps in Thin Edge JSON

### ISSUE THREAD

**Issue by albinsuresh** [thin-edge/thin-edge.io#2337]
**Is your feature request related to a problem? Please describe.**

Currently, Thin Edge JSON representation of measurements, events and alarms all accept a `time` field in ISO-8601 format. Although this representation is good for readability, it is not that easy to generate one, without using additional libraries. So, when simple child devices need to generate telemetry data, it'd be difficult for them to generate the time in this format, as they'd be forced to use external libraries.

**Describe the solution you'd like**

Using UNIX timestamp representation would be better for the following reasons:

1. Easy to generate natively on many platforms, without requiring any additional libs
2. Easy comparison of two timestamps
3. Parsing is easier (it is just a number)
4. Reduces overhead in messages

**Additional context**

Additional points to consider:

* Should we use seconds or milliseconds since 1970-01-01...I believe seconds might be more "standard" and expected
* Do we allow floats as the timestamps so that we can represent milliseconds and microseconds and how does floating points impact on precision (e.g. rounding errors)

**Comment by jarhodes314** [comment]
In #2508, I have started to implement support for parsing the relevant values. At the moment it parses an `i64` to a value in seconds, which is a pretty standard. In the description, I have summarised the various alternative representations we could use that spring to mind. We could add float support in addition to this for people that want to specify a greater precision than seconds. Equally, we could allow people to provide a string with an integer and then the relevant unit symbol (although this is more complicated and not something I would consider any sort of standard, but is in my mind quite simple) which would avoid rounding errors.

I'm not convinced rounding errors are much of an issue. After a bit of rough calculating, an `f64` has a 52 bit mantissa can store any integer less than or equal to 2^53 with perfect precision. Milliseconds are 1/1000 of a second, which is about 2^-10, so to represent values to millisecond precision* we could do that for values below 2^43 seconds = 278 737.322 years. If someone wants to guarantee their millisecond/microsecond/nanosecond precise values, they could always use RFC3339/ISO8601 instead.

* Ignoring precision errors that could come from converting a decimal value to binary
